### PR TITLE
Fix OpenGL in Mesa drivers

### PR DESCRIPTION
**Pull request by Zegeri**

### CODE REVIEW

File: src/xrEngine/Device_Initialize.cpp [hunk 1]
```diff
@@ -50,8 +50,16 @@ void CRenderDevice::Initialize()
         const Uint32 flags = SDL_WINDOW_BORDERLESS | SDL_WINDOW_HIDDEN |
             SDL_WINDOW_RESIZABLE | SDL_WINDOW_OPENGL;
 
-        m_sdlWnd = SDL_CreateWindow("S.T.A.L.K.E.R.: Call of Pripyat", 0, 0, 640, 480, flags);
+#ifdef USE_OGL
```
Comment (Xottab-DUTY) [inline src/xrEngine/Device_Initialize.cpp]: `USE_OGL` is defined and should only be used in `xrRender_GL` library.
In `Device`, it's better to check against against `psDeviceFlags`:

```cpp
if (psDeviceFlags.test(rsRGL))
{
	SDL_GL_SetAttribute(...);
	...
}
```

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]
Much thanks for the effort!

**Review by Xottab-DUTY — APPROVED** [review]

**Review by SkyLoaderr — APPROVED** [review]

File: src/xrEngine/Device_Initialize.cpp [hunk 2]
```diff
@@ -50,8 +50,17 @@ void CRenderDevice::Initialize()
         const Uint32 flags = SDL_WINDOW_BORDERLESS | SDL_WINDOW_HIDDEN |
             SDL_WINDOW_RESIZABLE | SDL_WINDOW_OPENGL;
 
-        m_sdlWnd = SDL_CreateWindow("S.T.A.L.K.E.R.: Call of Pripyat", 0, 0, 640, 480, flags);
+        if (psDeviceFlags.test(rsRGL))
```
Comment (Zegeri) [inline src/xrEngine/Device_Initialize.cpp:53]: Seems like at this point `psDeviceFlags` hasn't yet been initialized, so this test will fail always.

Comment (Xottab-DUTY) [reply src/xrEngine/Device_Initialize.cpp:53]: Oh.. True.. I'll think on it..

Comment (Xottab-DUTY) [reply src/xrEngine/Device_Initialize.cpp:53]: We can try to move initialization stages in `xrEngine/main.cpp`

**Pull request merged**